src/types.rs

```diff
@@ -157,10 +166,130 @@ impl std::fmt::Display for InstructionForm {
     }
 }
 
+// Register ---------------------------------------------------------------------------------------
+#[derive(Debug, Clone)]
+pub struct Register {
+    pub name: String,
+    pub alt_names: Vec<String>,
+    pub description: Option<String>,
+    pub reg_type: Option<RegisterType>,
+    pub width: Option<RegisterWidth>,
+    pub flag_info: Vec<RegisterBitInfo>,
+    pub arch: Option<Arch>,
+    pub url: Option<String>,
+}
+
+impl Hoverable for &Register {}
+
+impl Default for Register {
+    fn default() -> Self {
+        let name = String::new();
+        let alt_names = vec![];
+        let description = None;
+        let reg_type = None;
+        let width = None;
+        let flag_info = vec![];
+        let arch = None;
+        let url = None;
+
+        Self {
+            name,
+            alt_names,
+            description,
+            reg_type,
+            width,
+            flag_info,
+            arch,
+            url,
+        }
+    }
+}
+
+impl std::fmt::Display for Register {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        // basic fields
+        let header: String;
+        if let Some(arch) = &self.arch {
+            header = format!("{} [{}]", &self.name.to_uppercase(), arch.as_ref());
+        } else {
+            header = self.name.to_uppercase();
+        }
+
+        let mut v: Vec<&str> = if let Some(description_) = &self.description {
+            vec![&header, description_, "\n"]
+        } else {
+            vec![&header, "\n"]
+        };
+
+        // Register Type
+        let reg_type;
+        if let Some(reg_type_) = &self.reg_type {
+            reg_type = format!("Type: {}", reg_type_);
+            v.push(reg_type.as_str());
+        }
+
+        // Register Width
+        let reg_width;
+        if let Some(reg_width_) = &self.width {
+            reg_width = format!("Width: {}", reg_width_);
+            v.push(reg_width.as_str());
+        }
+
+        // Bit-mask flag meanings if applicable
+        let flag_heading = "\n## Flags:";
```

Comment:
Do these flags work? is there any register in the files of `samples/` that I can use  to test this with?

Adding a [`gs` word](https://github.com/WillLillis/asm-lsp/blob/c7f89a684732b4d6555ba0113fe2756e17257411/registers/x86.xml#L609) inside the `samples/gas.s` and hovering over it doesn't seem to show  the Flags section.

Reply:
Sorry what do you mean by adding a gs word to the sample file? My original intention was for the flag information to be displayed if the user just hovered over the relevant register, kind of similar to the "Forms" fields in the instruction hovers. For example if the user was using register ``%idtr`` in an instruction or had the name written in a comment they could pull the information up. E.g.:
Added comment on line 27:
![image](https://github.com/bergercookie/asm-lsp/assets/87077023/a80e9bab-8281-406e-b8f4-ab74a15808ad)
Hover on ``%idtr`` from comment
![image](https://github.com/bergercookie/asm-lsp/assets/87077023/5e7253df-2788-4da4-9c0c-937192e81c0e)

Reply:
Yeah, sorry I think I messed up the `eflags` and `gs` registers. It works fine!